Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -35,32 +35,110 @@ import TensorFlow
 /// [2]: https://github.com/borglab/gtsam/blob/develop/doc/math.pdf
 /// [3]: Actually, we define the pullbacks because Swift doesn't support differentials very well
 ///      yet.
-public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
-  /// The pose's rotation.
-  ///
-  /// This is the projection SE(2) -> SO(2).
-  @differentiable public var rot: Rot2 { rotStorage }
+public struct Pose2: Manifold, Equatable, TangentStandardBasis, KeyPathIterable {
+  // MARK: - Manifold conformance
 
-  /// The pose's translation.
-  ///
-  /// This is the projection SE(2) -> R^2. (Note: not a group homomorphism!)
-  @differentiable public var t: Vector2 { tStorage }
+  public var coordinateStorage: Pose2Coordinate
+  public init(coordinateStorage: Pose2Coordinate) { self.coordinateStorage = coordinateStorage }
+
+  public mutating func move(along direction: Coordinate.LocalCoordinate) {
+    coordinateStorage = coordinateStorage.global(direction)
+  }
 
   /// Creates a `Pose2` with rotation `r` and translation `t`.
   ///
   /// This is the bijection SO(2) x R^2 -> SE(2), where "x" means direct product of groups. (Note:
   /// not a group homomorphism!)
   @differentiable
   public init(_ r: Rot2, _ t: Vector2) {
-    tStorage = t
-    rotStorage = r
+    self.init(coordinate: Pose2Coordinate(r, t))
+  }
+  
+  // MARK: Convenience Attributes
+  
+  @differentiable public var t: Vector2 { coordinate.t }
+  
+  @differentiable public var rot: Rot2 { coordinate.rot }
+  
+  /// Twist coordinates, with the first component for rotation and the remaining components for
+  /// translation.
+  ///
+  /// See the documentation comment on the declaration of `Pose2` for more information.
+  public typealias TangentVector = Vector3
```

Comment:
This should not be necessary. It is automatically inferred to be equal to Pose2Coordinate.LocalCoordinate.

Reply:
Thanks!